### PR TITLE
Fix softmax output backprop, plus tests!

### PR DESCRIPTION
**Pull request by mbrookhart**

### CODE REVIEW

**Comment by mbrookhart** [comment]
I'm honestly not sure, I'll do some debugging in the morning and take it out of WIP once I'm sure.

File: src/ngraph/ngraph_emitter.cc [hunk 1]
```diff
@@ -1509,10 +1511,19 @@ void Emitter::CreateLossOps() {
     auto gradient = softmax - label;
 
     if (ignore) {
+      ngraph::Shape new_shape(gradient->get_shape().size(), 1);
```
Comment (louisfeng) [inline src/ngraph/ngraph_emitter.cc]: Could you please add a comment here about reshaping `mask` to allow broadcasting to the shape of gradient.
